File: コードレビュー/05.マスター/プログラム作成問題：マスター編/20200210_173738496/src/ネームバトラー/Magic.java
package ネームバトラー;

public class Magic {
	
	protected String name;
	protected int useMP;
	
	public Magic(String name,int useMP) {
		this.name=name;
		this.useMP=useMP;
	}
	
	public String GetName() {
		return this.name;
	}
	
	public int GetUseMP() {
		return this.useMP;
	}
	
	public void UseMagic(Player attacker,Player affected) {
		
	}
}

File: コードレビュー/07.Unityアプリ/20200701_USSBB1M39/Assets/Script/Magic.cs
﻿using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Magic
{

	// =======================
	// フィールド変数
	// =======================
	// 魔法の名前
	protected string name;
	// 消費MP
	protected int usemp;


	// =======================
	// コンストラクタ
	// =======================
	public Magic()
	{

	}

	// =======================
	// Getter / Setter
	// =======================
	public string GetName()
	{
		return this.name;
	}

	public int GetUseMP()
	{
		return this.usemp;
	}

	// =======================
	// protected メソッド
	// =======================

	// =======================
	// private メソッド
	// =======================

	// =======================
	// public メソッド
	// =======================
	/**
	 *魔法を行使する
	 * @param activePlayer : 行為プレイヤー
	 * @param passivePlayer : 対象プレイヤー
	 */
	public virtual void effect(Player activePlayer, Player passivePlayer)
	{
		// 魔法ごとに処理をオーバーライドして記述する
	}


}